sdk/azidentity/chained_token_credential.go

```diff
@@ -14,15 +14,20 @@ import (
 
 // ChainedTokenCredentialOptions contains optional parameters for ChainedTokenCredential.
 type ChainedTokenCredentialOptions struct {
-	// placeholder for future options
+	// If true, it will not assume the first successful credential should be always used.
+	RetryAllSources bool
```

Comment:
Is this really a retry of all sources?  It still stops at the first succesfull match.  Would `RetrySources` make more sense

Reply:
Good thinking! I like RetrySources!

Reply:
Pushed the change here: https://github.com/Azure/azure-sdk-for-go/pull/16392/commits/e288440b96d230c971216758a8bc5881d9241141